camayoc/tests/remote/rho/test_scan.py

```diff
@@ -96,21 +96,34 @@ def test_scan(isolated_filesystem, profile):
             known_facts = {}
             for host in cfg['rho']['hosts']:
                 # find the facts for the scanned host we are inspecting
-                if host['ip'] == row['connection.host']:
+                if host['ip'] == row.get('connection.host'):
                     known_facts = host['facts']
                     break
-            for fact in known_facts.keys():
+            missed_facts = known_facts.keys() - row.keys()
+            if (missed_facts != set()):
+                # This means there are facts that we expected,
+                # but were not collected
+                for fact in missed_facts:
```

Comment:
The above if block is not needed the for can be used directly. If the missed_facts is empty no iteration will run.